mono/utils/memfuncs.c

```diff
@@ -288,7 +289,33 @@ mono_determine_physical_ram_size (void)
 		return 134217728;
 	}
 
-	return page_size * num_pages;
+	memsize = page_size * num_pages;	/* Calculate physical memory size */
+
+	gint64 restricted_limit = getRestrictedPhysicalMemoryLimit();	/* Check for any cgroup limit */
+	if (restricted_limit != 0) {
+		gchar *heapHardLimit = getenv("DOTNET_GCHeapHardLimit");	/* See if user has set a limit */
```

Comment:
this should also check for `COMPlus_GCHeapHardLimit` (old name for the env var).

Reply:
Done